toolkit/scripts/analysis.mk

```diff
@@ -86,3 +89,49 @@ sodiff-check: $(BUILT_PACKAGES_FILE) | $(SODIFF_REPO_FILE)
 	<$(BUILT_PACKAGES_FILE) $(SODIFF_SCRIPT) $(RPMS_DIR)/ $(SODIFF_REPO_FILE) $(RELEASE_MAJOR_ID) $(SODIFF_OUTPUT_FOLDER)
 
 package-toolkit: $(SODIFF_REPO_FILE)
+
+######## LICENSE CHECK ########
+
+license_check_build_dir   = $(BUILD_DIR)/license_check_tool
+license_out_dir           = $(OUT_DIR)/license_check
+license_results_file_pkg  = $(license_out_dir)/license_check_results_pkg.json
+license_results_file_imge = $(license_out_dir)/license_check_results_image_$(config_name).json
+
+.PHONY: license-check license-check-img clean-license-check
+
+clean: clean-license-check
+clean-license-check:
+	@echo Verifying no mountpoints present in $(license_check_build_dir)
+	$(SCRIPTS_DIR)/safeunmount.sh "$(license_check_build_dir)" && \
+	rm -rf $(license_check_build_dir) && \
+	rm -rf $(license_out_dir)
+
+# If we are using the default RPMS_DIR as LICENSE_CHECK_DIRS, we are responsible for building the rpms as needed.
+# Can set REBUILD_PACKAGES=n to skip rebuilding the rpms if they are already built.
+ifeq ($(LICENSE_CHECK_DIRS),$(RPMS_DIR))
+license-check: $(LICENSE_CHECK_DIRS)
+endif
+
+##help:target:license-check=Validate all packages in the RPMS_DIR for license compliance. Set LICENSE_CHECK_DIRS to override source.
+license-check: $(go-licensecheck) $(chroot_worker) $(LICENSE_CHECK_EXCEPTION_FILE) $(LICENSE_CHECK_NAME_FILE)
+	$(go-licensecheck) \
+		$(foreach license_dir, $(LICENSE_CHECK_DIRS),--rpm-dirs="$(license_dir)" ) \
```

Comment:
Can we shift the logic of handling the list into the tool
`$(foreach license_dir, $(LICENSE_CHECK_DIRS),--rpm-dirs="$(license_dir)" )`
we just pass the entire list of license_check_dirs to the tool.
This will help us with our goal of making Makefiles leaner wrt logic

Reply:
I think that is the wrong direction here. The argument parsing library we use has explicit code to take lists of directories etc. I think being scared of basic makefile functionality is the wrong way to go about this. We should try to minimize the complex logic, but in this case we should leverage the error checking intrinsic in the arg parsing library.